examples/c/tc.c

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2022 Hengqi Chen */
+#include <signal.h>
+#include <unistd.h>
+#include "tc.skel.h"
+
+#define LO_IFINDEX	1
+
+static volatile sig_atomic_t exiting = 0;
+
+static void sig_int(int signo)
+{
+	exiting = 1;
+}
+
+static int libbpf_print_fn(enum libbpf_print_level level, const char *format, va_list args)
+{
+	return vfprintf(stderr, format, args);
+}
+
+int main(int argc, char **argv)
+{
+	DECLARE_LIBBPF_OPTS(bpf_tc_hook, tc_hook,
+		.ifindex = LO_IFINDEX, .attach_point = BPF_TC_INGRESS);
```

Comment:
I recommend using the following function to decide the targeted interface, since the index may vary on different devices.

```c
u_int32_t lo_ifindex = if_nametoindex("lo");
```
The function declaration is in `net/if.h`.

Reply:
The Linux kernel ensures that loopback device is always the first network device, even in network namespace.

Reply:
> The Linux kernel ensures that loopback device is always the first network device, even in network namespace.

Got it. Thanks.